cmds.go

```diff
@@ -1,25 +1,122 @@
 package main
 
-import "github.com/Azure/azure-docker-extension/pkg/vmextension"
+import (
+	"os"
+	"path/filepath"
+	"strconv"
 
-type cmdFunc func(vmextension.HandlerEnvironment) error
+	"github.com/Azure/azure-docker-extension/pkg/vmextension"
+	"github.com/Azure/custom-script-extension-linux/seqnum"
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+)
+
+type cmdFunc func(*log.Context, vmextension.HandlerEnvironment) error
+
+type cmd struct {
+	f                  cmdFunc // associated function
+	name               string  // human readable string
+	shouldReportStatus bool    // determines if running this should log to a .status file
+}
 
 var (
-	cmds = map[string]struct {
-		f                  cmdFunc // associated function
-		name               string  // human readable string
-		shouldReportStatus bool    // determines if running this should log to a .status file
-	}{
-		"install":   {install, "Install", false},
-		"uninstall": {uninstall, "Uninstall", false},
-		"enable":    {enable, "Enable", true},
-		"update":    {update, "Update", true},
-		"disable":   {disable, "Disable", true},
+	cmdInstall   = cmd{install, "Install", false}
+	cmdEnable    = cmd{enable, "Enable", true}
+	cmdUninstall = cmd{uninstall, "Uninstall", false}
+
+	cmds = map[string]cmd{
+		"install":   cmdInstall,
+		"uninstall": cmdUninstall,
+		"enable":    cmdEnable,
+		"update":    {noop, "Update", true},
+		"disable":   {noop, "Disable", true},
 	}
 )
 
-func install(vmextension.HandlerEnvironment) error   { return nil }
-func uninstall(vmextension.HandlerEnvironment) error { return nil }
-func enable(vmextension.HandlerEnvironment) error    { return nil }
-func update(vmextension.HandlerEnvironment) error    { return nil }
-func disable(vmextension.HandlerEnvironment) error   { return nil }
+func noop(ctx *log.Context, h vmextension.HandlerEnvironment) error {
+	ctx.Log("event", "noop")
+	return nil
+}
+
+func install(ctx *log.Context, h vmextension.HandlerEnvironment) error {
+	if err := os.MkdirAll(dataDir, 0755); err != nil {
+		return errors.Wrap(err, "failed to create data dir")
+	}
+	ctx.Log("event", "created data dir", "path", dataDir)
+	ctx.Log("event", "installed")
+	return nil
+}
+
+func enable(ctx *log.Context, h vmextension.HandlerEnvironment) error {
+	// parse the extension handler settings (not available prior to 'enable')
+	cfg, err := parseAndValidateSettings(ctx, h.HandlerEnvironment.ConfigFolder)
+	if err != nil {
+		return errors.Wrap(err, "failed to get configuration")
+	}
+
+	seq, err := strconv.Atoi(h.SeqNo)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse seqnum: %q", h.SeqNo)
+	}
+
+	// exit if this sequence number (a snapshot of the configuration) is
+	// processed. if not, save this sequence number before proceeding.
+	sf := filepath.Join(dataDir, seqNumFile)
+	ctx.Log("event", "comparing seqnum")
+	if b, err := seqnum.IsSmallerOrEqualThan(sf, seq); err != nil {
```

Comment:
This if statement feels like a low-level concept mixed into a high-level concept.  The semantic meaning of IsSmallerOrEqualThan means much less than IsConfigProcessed() (<insert better name>).  If you pick a good subroutine name you can remove the comment.